reboot_toolkit/toolkit_backend.py

```diff
@@ -259,27 +264,46 @@ def list_available_s3_keys(org_id: str, df: pd.DataFrame) -> list[str]:
 
     return all_files
 
-
 def load_games_to_df_from_s3_paths(
-        game_paths: list[str], add_ik_joints: bool = False, add_elbow_var_val: bool = False
+        game_paths: list[str], add_ik_joints: bool = False, add_elbow_var_val: bool = False, game_proportion: float = 1.0

```

Comment:
I would actually even take `game_proportion` out of the function call and hard code it in the body of the function. Just to prevent someone from loading like an entire season's worth of games.

In the function itself, you could have two different `game_number_threshold` values in this way...

```
if len(game_paths) > game_number_threshold_for_sampling_games:
    game_paths = random sample of game paths

if len(game_paths) > game_number_threshold_for_sampling_movements:
    movement_proportion = 0.2
```

Reply:
I ended up changing the order to make sure that we check the number of games first to set the `movement_proportion`. The problem with setting the `game_proportion` is that it would prevent us from ever sampling a certain percentage of movements from every game (which is what we are trying to do at least in this case) and could be problematic for hitting if we even try to pull two weeks of games which could be greater than 10 games. I doubt that we would need to sim every game but since this loading function is now integrated, it would affect the metrics and analysis notebooks as well.

So I think there are a few possible options:

- Set a hard-coded game sampling proportion if the number of games, acknowledging that it would prevent every game to be sampled.
- Have a different threshold for hitting and pitching
- Have multiple `movement_proportions` depending on how many games are loaded. Example 20% if more than 10, 10% if more than 20

Thoughts? Since my understanding of the intent for this specific use case was to sample a certain percentage from every game, I'll leave this for the moment and run the notebook from the branch (unless you think otherwise), then we can plan to add in the automatic game sampling when we merge with main.